src/analyze/expr.rs

```diff
@@ -58,11 +58,19 @@ impl PureAnalyzer {
                 let struct_type = match &inner.ctype {
                     Type::Pointer(ctype, _) => match &**ctype {
                         Type::Union(_) | Type::Struct(_) => (**ctype).clone(),
+                        Type::Error => {
+                            // already type error
+                            return inner;
+                        }
```

Comment:
I think this is pretty clear from the code.
```suggestion
                        Type::Error => return inner,
```